src/System.Data.SqlClient/src/System/Data/Sql/SqlFunctionAttribute.cs

```diff
@@ -0,0 +1,131 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.SqlServer.Server
+{
+    [Serializable]
+    public enum DataAccessKind
+    {
+        None = 0,
+        Read = 1,
+    }
+
+    [Serializable]
+    public enum SystemDataAccessKind
+    {
+        None = 0,
+        Read = 1,
+    }
+
+    // sql specific attribute
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false), Serializable]
+    public class SqlFunctionAttribute : Attribute
+    {
+        private bool _isDeterministic;
+        private DataAccessKind _dataAccess;
+        private SystemDataAccessKind _systemDataAccess;
+        private bool _isPrecise;
+        private string _name;
+        private string _tableDefinition;
+        private string _fillRowMethodName;
+
+        public SqlFunctionAttribute()
+        {
+            // default values
+            _isDeterministic = false;
+            _dataAccess = DataAccessKind.None;
+            _systemDataAccess = SystemDataAccessKind.None;
+            _isPrecise = false;
+            _name = null;
+            _tableDefinition = null;
+            _fillRowMethodName = null;
+        }
+
+        public bool IsDeterministic
```

Comment:
Nit: These properties can be written in Short hand using `=>`  
```
public bool IsDeterministic 
{
   get => _isDeterministic;
   set => _isDeterministic = value;
}
```

Applies to all properties.